File: native-ios/SaintCharbelApp/Core/AppTheme.swift
import SwiftUI

enum AppTheme {
    static let backgroundTop = Color(red: 0.07, green: 0.10, blue: 0.14)
    static let backgroundBottom = Color(red: 0.03, green: 0.05, blue: 0.08)
    static let panel = Color(red: 0.10, green: 0.14, blue: 0.19)
    static let chromeBackground = Color(red: 0.06, green: 0.08, blue: 0.12)
    static let chromeBorder = Color.white.opacity(0.10)
    static let panelBorder = Color.white.opacity(0.12)
    static let textPrimary = Color(red: 0.96, green: 0.94, blue: 0.89)
    static let textSecondary = Color(red: 0.79, green: 0.80, blue: 0.82)
    static let tabInactive = Color(red: 0.72, green: 0.74, blue: 0.78)
    static let gold = Color(red: 0.84, green: 0.72, blue: 0.44)
    static let ember = Color(red: 0.71, green: 0.47, blue: 0.32)
    static let olive = Color(red: 0.48, green: 0.58, blue: 0.40)
    static let rose = Color(red: 0.63, green: 0.38, blue: 0.44)
    static let paper = Color(red: 0.95, green: 0.91, blue: 0.84)
    static let paperShadow = Color(red: 0.83, green: 0.73, blue: 0.58)
    static let ink = Color(red: 0.18, green: 0.18, blue: 0.20)
    static let storyBlue = Color(red: 0.32, green: 0.46, blue: 0.67)
    static let storyBerry = Color(red: 0.65, green: 0.32, blue: 0.39)

    static let backgroundGradient = LinearGradient(
        colors: [backgroundTop, backgroundBottom],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )

    static let storybookGradient = LinearGradient(
        colors: [
            Color(red: 0.09, green: 0.13, blue: 0.21),
            Color(red: 0.18, green: 0.22, blue: 0.33),
            Color(red: 0.27, green: 0.20, blue: 0.15)
        ],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )

    static func accent(for set: RosarySet) -> Color {
        switch set {
        case .joyful:
            return gold
        case .luminous:
            return Color(red: 0.56, green: 0.76, blue: 0.78)
        case .sorrowful:
            return rose
        case .glorious:
            return olive
        }
    }

    static func gradient(for set: RosarySet) -> LinearGradient {
        switch set {
        case .joyful:
            return LinearGradient(colors: [gold.opacity(0.7), ember.opacity(0.65)], startPoint: .topLeading, endPoint: .bottomTrailing)
        case .luminous:
            return LinearGradient(colors: [Color(red: 0.54, green: 0.77, blue: 0.80), Color(red: 0.35, green: 0.48, blue: 0.73)], startPoint: .topLeading, endPoint: .bottomTrailing)
        case .sorrowful:
            return LinearGradient(colors: [rose.opacity(0.82), Color(red: 0.30, green: 0.18, blue: 0.24)], startPoint: .topLeading, endPoint: .bottomTrailing)
        case .glorious:
            return LinearGradient(colors: [olive.opacity(0.92), Color(red: 0.29, green: 0.45, blue: 0.34)], startPoint: .topLeading, endPoint: .bottomTrailing)
        }
    }
}

File: native-android/app/src/main/java/com/jnrahme/androidsaintcharbel/core/theme/AppTheme.kt
package com.jnrahme.androidsaintcharbel.core.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import com.jnrahme.androidsaintcharbel.core.model.RosarySet

object AppPalette {
    val BackgroundTop = Color(0xFF121A24)
    val BackgroundBottom = Color(0xFF08101A)
    val Panel = Color(0xFF1A2431)
    val ChromeBackground = Color(0xFF0F141E)
    val ChromeBorder = Color(0x1AFFFFFF)
    val PanelBorder = Color(0x1FFFFFFF)
    val TextPrimary = Color(0xFFF5F0E3)
    val TextSecondary = Color(0xFFC8CDD1)
    val TabInactive = Color(0xFFB8BCC6)
    val Gold = Color(0xFFD8B96E)
    val Ember = Color(0xFFB67852)
    val Olive = Color(0xFF7A9466)
    val Rose = Color(0xFFA36474)
    val Paper = Color(0xFFF1E8D5)
    val PaperShadow = Color(0xFFD5BE95)
    val Ink = Color(0xFF2D2D33)
    val StoryBlue = Color(0xFF5376AA)
    val StoryBerry = Color(0xFFA55264)

    val BackgroundBrush: Brush = Brush.linearGradient(
        colors = listOf(BackgroundTop, BackgroundBottom),
    )

    val StorybookBrush: Brush = Brush.linearGradient(
        colors = listOf(
            Color(0xFF172136),
            Color(0xFF2F394F),
            Color(0xFF473526),
        ),
    )

    fun accent(set: RosarySet): Color = when (set) {
        RosarySet.JOYFUL -> Gold
        RosarySet.LUMINOUS -> Color(0xFF8FC2C8)
        RosarySet.SORROWFUL -> Rose
        RosarySet.GLORIOUS -> Olive
    }

    fun gradient(set: RosarySet): Brush = Brush.linearGradient(
        colors = when (set) {
            RosarySet.JOYFUL -> listOf(Gold.copy(alpha = 0.7f), Ember.copy(alpha = 0.65f))
            RosarySet.LUMINOUS -> listOf(Color(0xFF8AC4CC), Color(0xFF5E7BB9))
            RosarySet.SORROWFUL -> listOf(Rose.copy(alpha = 0.82f), Color(0xFF4D2D3A))
            RosarySet.GLORIOUS -> listOf(Olive.copy(alpha = 0.92f), Color(0xFF4A6A53))
        },
    )
}

private val SaintCharbelColorScheme = darkColorScheme(
    primary = AppPalette.Gold,
    secondary = AppPalette.TextSecondary,
    background = AppPalette.BackgroundBottom,
    surface = AppPalette.Panel,
    onPrimary = AppPalette.Ink,
    onSecondary = AppPalette.TextPrimary,
    onBackground = AppPalette.TextPrimary,
    onSurface = AppPalette.TextPrimary,
)

private val SaintCharbelTypography = androidx.compose.material3.Typography(
    displayLarge = TextStyle(fontFamily = FontFamily.Serif, fontWeight = FontWeight.Bold, fontSize = 40.sp, lineHeight = 44.sp),
    displayMedium = TextStyle(fontFamily = FontFamily.Serif, fontWeight = FontWeight.Bold, fontSize = 34.sp, lineHeight = 38.sp),
    headlineLarge = TextStyle(fontFamily = FontFamily.Serif, fontWeight = FontWeight.Bold, fontSize = 32.sp, lineHeight = 36.sp),
    headlineMedium = TextStyle(fontFamily = FontFamily.Serif, fontWeight = FontWeight.SemiBold, fontSize = 28.sp, lineHeight = 32.sp),
    titleLarge = TextStyle(fontWeight = FontWeight.SemiBold, fontSize = 22.sp, lineHeight = 28.sp),
    titleMedium = TextStyle(fontWeight = FontWeight.SemiBold, fontSize = 18.sp, lineHeight = 24.sp),
    bodyLarge = TextStyle(fontSize = 16.sp, lineHeight = 24.sp),
    bodyMedium = TextStyle(fontSize = 15.sp, lineHeight = 22.sp),
    labelLarge = TextStyle(fontWeight = FontWeight.SemiBold, fontSize = 14.sp, lineHeight = 20.sp),
)

@Composable
fun SaintCharbelTheme(content: @Composable () -> Unit) {
    val dark = isSystemInDarkTheme()
    MaterialTheme(
        colorScheme = SaintCharbelColorScheme,
        typography = SaintCharbelTypography,
        content = content,
    )
}

